### PR TITLE
SEQNG-1268 Log reasons to change TCS configuration.

### PR DESCRIPTION
**Pull request by jluhrs**
Log every EPICS parameter changed in TCS. The log message has the parameter name, old value, and new value.
The new messages are output only if seqexec.server.tcs.trace is defined.

### CODE REVIEW

File: modules/server/src/main/scala/seqexec/server/tcs/TcsControllerEpicsCommon.scala [hunk 1]
```diff
@@ -239,64 +245,94 @@ object TcsControllerEpicsCommon {
     demand:  T,
     act:     T => F[Unit],
     lens:    Lens[C, T]
-  ): Option[C => F[C]] =
-    (used && current =!= demand).option(c => act(demand) *> lens.set(demand)(c).pure[F])
+  )(name:    String): Option[WithDebug[C => F[C]]] =
```
Comment (cquiroz) [inline modules/server/src/main/scala/seqexec/server/tcs/TcsControllerEpicsCommon.scala:248]: Would it make sense to make a `WithDebug` typeclass and attach it to `F` like `Logger` or natchez
https://github.com/tpolecat/natchez

Comment (jluhrs) [reply modules/server/src/main/scala/seqexec/server/tcs/TcsControllerEpicsCommon.scala:248]: Do you mean, like to be able to use an `F: Applicative` or `F: Monad`, and define a way to combine the debug messages at the same time you do `map` or `flatMap` on the `F` ?

Comment (cquiroz) [reply modules/server/src/main/scala/seqexec/server/tcs/TcsControllerEpicsCommon.scala:248]: Yeah use it like `F: WithDebug` but it is just an algebra with methods like `_.withDebug`

Comment (jluhrs) [reply modules/server/src/main/scala/seqexec/server/tcs/TcsControllerEpicsCommon.scala:248]: I don't think that applies that way. `withDebug` is really a constructor that takes an `A` and produces a `WithDebug[A]`.  There is no point on having and `x: F` (with `F: WithDebug`) just to be able to call `x.withDebug`

**Comment by cquiroz** [comment]
How are these changes going to be included in `observe`?

**Review by cquiroz — APPROVED** [review]
LGTM

**Pull request merged**